File: server/Server/Models/Domain/Asset.cs
using server.Models.Domain.Enums;

namespace server.Models.Domain;

public class Asset
{
    public int AssetId { get; set; }
    public string ExternalId { get; set; } = string.Empty;
    public string Filename { get; set; } = string.Empty;
    public string? MimeType { get; set; }
    public long? SizeBytes { get; set; }
    public int? Width { get; set; }
    public int? Height { get; set; }
    public int? DurationMs { get; set; }
    public string? Metadata { get; set; }
    public AssetStatus Status { get; set; }
    public DateTime? DeletedAt { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Foreign Keys
    public int ProjectId { get; set; }
    public int DataSourceId { get; set; }

    // Navigation Properties
    public virtual Project Project { get; set; } = null!;
    public virtual DataSource DataSource { get; set; } = null!;

    public virtual ICollection<Task> Tasks { get; set; } = [];
    public virtual ICollection<Annotation> Annotations { get; set; } = [];
    public virtual ICollection<Issue> Issues { get; set; } = [];
}

File: frontend/src/types/asset/asset.ts
export enum AssetStatus {
    PENDING_IMPORT = 'PENDING_IMPORT',
    IMPORTED = 'IMPORTED',
    IMPORT_ERROR = 'IMPORT_ERROR',
    PENDING_PROCESSING = 'PENDING_PROCESSING',
    PROCESSING = 'PROCESSING',
    PROCESSING_ERROR = 'PROCESSING_ERROR',
    READY_FOR_ANNOTATION = 'READY_FOR_ANNOTATION',
    ANNOTATION_IN_PROGRESS = 'ANNOTATION_IN_PROGRESS',
    ANNOTATED = 'ANNOTATED',
    PENDING_REVIEW = 'PENDING_REVIEW',
    REVIEW_IN_PROGRESS = 'REVIEW_IN_PROGRESS',
    REVIEW_ACCEPTED = 'REVIEW_ACCEPTED',
    REVIEW_REJECTED = 'REVIEW_REJECTED',
    EXPORTED = 'EXPORTED',
    ARCHIVED = 'ARCHIVED',
}

export interface Asset {
    assetId: number;
    projectId: number;
    dataSourceId: number;
    durationMs: number;
    externalId: string;
    filename: string;
    height: number;
    width: number;
    mimeType: string;
    sizeBytes: number;
    status: AssetStatus;
    createdAt: string
    updatedAt: string;
}